airflow/ti_deps/dep_context.py

```diff
@@ -85,23 +84,16 @@ def __init__(
         self.ignore_ti_state = ignore_ti_state
         self.finished_tasks = finished_tasks
 
-    def ensure_finished_tasks(self, dag, execution_date: pendulum.DateTime, session: Session):
+    def ensure_finished_tasks(self, dag_run, session: Session):
```

Comment:
```suggestion
    def ensure_finished_tasks(self, dag_run: "DagRun", session: Session):
```

(And an import of course.)